src/main/scala/ie/zalando/fabric/gateway/service/StackSetOperations.scala

```diff
@@ -20,7 +22,27 @@ class StackSetOperations(k8sClient: KubernetesClient)(implicit execCtxt: Executi
 
   private type StackSet = CustomResource[StackSetSpec, StackSetStatus]
 
-  implicit val stackSvcFmt: Format[StackDefinedService] = Json.format[StackDefinedService]
+  implicit val servicePortReads: Reads[K8sServicePortIdentifier] = {
+    case JsString(s) => JsSuccess(NamedServicePort(s))
+    case other => Reads.IntReads.reads(other).map(NumericServicePort)
+  }
+  implicit val servicePortWrites: Writes[K8sServicePortIdentifier] = {
+    case NamedServicePort(name) => JsString(name)
+    case NumericServicePort(port) => JsNumber(port)
+  }
+  implicit val stackSvcReads: Reads[StackDefinedService] = (
```

Comment:
Crap that stackSvcReads and stackSvcWrites are still necessary after the K8sServicePortIdentifier read and writer is defined, lucky it's not used in a lot of places

Reply:
Should it not be picked up as an implicit if it's defined before `Json.format[StackDefinedService]`
?

Reply:
How would it know to treat `servicePort` differently?

Reply:
Does it not use the implicit `Reads[K8sServicePortIdentifier]` in `Json.format[StackDefinedService]` when defined before it?

Reply:
@irishshagua think I know what you mean if you were commenting on a different line. See:
https://github.com/zalando-incubator/fabric-gateway/pull/31/commits/7f7c1e88b5486a39c78d8812f01bdb1d29c7113a

Reply:
Why are `stackSvcReads` and `stackSvcWrites` needed now?

Reply:
Because we need it to configure it to handle the Json property `servicePort` in a custom way

Reply:
That worked, much nicer. 👍